collect_app/src/main/java/org/odk/collect/android/preferences/AdminPreferencesFragment.java

```diff
@@ -144,36 +144,31 @@ public boolean onPreferenceClick(Preference preference) {
         return false;
     }
 
-    public static class MainMenuAccessPreferences extends BasePreferenceFragment {
-
+    public static class MainMenuAccessPreferences extends PreferenceFragmentCompat {
         @Override
-        public void onCreate(Bundle savedInstanceState) {
-            super.onCreate(savedInstanceState);
+        public void onCreatePreferences(Bundle savedInstanceState, String rootKey) {
             getPreferenceManager().setSharedPreferencesName(ADMIN_PREFERENCES);
-
-            addPreferencesFromResource(R.xml.main_menu_access_preferences);
+            setPreferencesFromResource(R.xml.main_menu_access_preferences, rootKey);
+            ((CollectAbstractActivity) getActivity()).initToolbar(getPreferenceScreen().getTitle());
```

Comment:
Conditional cast for the child Fragments as well.

Reply:
Now it doesn't exist there and we set title only in BasePreferenceFragment which works for every screen.